src/components/Offers/Form/form-components/OfferForm.js

```diff
@@ -102,12 +102,36 @@ const OfferForm = ({ context, title }) => {
         },
     };
 
+    useEffect(() => {
+        if (Object.keys(errors).length !== 0) {
+            const element = document.getElementById(Object.keys(errors)[0]);
+            if (element?.scrollIntoView) {
+                element.scrollIntoView({ behavior: "smooth" });
+            }
+            if (element?.focus) {
+                element.focus();
+            }
+        }
+    }, [errors]);
+
+    useEffect(() => {
+        if (Object.keys(requestErrors).length !== 0) {
+            const element = document.getElementById(Object.keys(requestErrors)[0]);
+            if (element?.scrollIntoView) {
+                element.scrollIntoView({ behavior: "smooth" });
+            }
+            if (element?.focus) {
+                element.focus();
+            }
+        }
+    }, [requestErrors]);
```

Comment:
```suggestion
    useEffect(() => {
        if (Object.keys(errors).length !== 0) {
            let element = document.getElementById(Object.keys(errors)[0]);
            if (!element) {
                element = document.getElementById(Object.keys(requestErrors)[0]);
            }
            if (element?.scrollIntoView) {
                element.scrollIntoView({ behavior: "smooth" });
            }
            if (element?.focus) {
                element.focus();
            }
        }
    }, [errors, requestErrors]);
```
I think something like this is simpler, there's a lot of repeated logic here. I didn't test it so make sure the code works

Reply:
I tried something similar while developing and it does not work well because, for some reason, `errors` and `requestErrors` are both updated when submitting but not at the same time (one first and immediately the other after) and it results in activating the scroll when it shouldn't. For example, when submitting successfully, it would scroll up while transitioning to the new page.

I see that, in your version, you don't check the length of `requestErrors`, so it may not work properly with such errors (like setting the publication date in the past).

Reply:
I understand, maybe we could extract the code to an outside function that takes the respective array as a param. What do you think?

Reply:
Yeah, I agree!